### ISSUE TITLE
Password-protect PDFs

### ISSUE THREAD

**Issue by MarkusFF** [typst/typst#4387]
### Description

PDFs have encryption support that allow you to password-protect aspects of the PDF to prevent e.g. reading, modification, printing etc. On some tex flavours this could be achieved with, for example:

```latex
\special{pdf:encrypt ownerpw (abc) userpw (xyz) length 128 perm 2052}
```

This is a feature widely used in PDFs that are sold, such as research papers, standards documents and so on. Although frequently overused this is likely an essential feature for some users.

It should be noted that a PDF can already be modified in this way after generation, for example with pdftk

### Use Case

- Prevent modifications on certificates generated with typst (by e.g. changing the date of calibration of an instrument)
- Protect sensitive content from being viewed without a password

**Comment by JamesxX** [comment]
I imagine this would make for a good cli argument (it deals only with pdf exports so I don't think it should belong in document source itself)

**Comment by Enivex** [comment]
> This is a feature widely used in PDFs that are sold, such as research papers, 
I can't recall ever having seen a password protected paper. They're hidden behind a paywall, but the actual pdf is not protected in any way.

**Comment by JamesxX** [comment]
> > This is a feature widely used in PDFs that are sold, such as research papers,
> 
> I can't recall ever having seen a password protected paper. They're hidden behind a paywall, but the actual pdf is not protected in any way.

I've mostly seen it when dealing with conference stuff

**Comment by asmaier** [comment]
As a workaround you can simply zip the PDF and set a password for the zip file.

**Comment by mohe2015** [comment]
The issue author already said there are tools that can add a password to a PDF so I think that is not an issue. This is about integrating such functionality into Typst.

**Comment by reknih** [comment]
I don't think we plan to ship this. Security is always a big can of worms and with PDF, the security landscape generally evolves faster than both the standards and viewers. Just take a look at PDF 1.7 extension levels: Many of them mainly introduced new ciphers and the like. We do not want to focus efforts on ensuring that we can ship a safe & secure feature here, it's better left up to dedicated tools.
